docs/headers.md

```diff
@@ -91,16 +91,16 @@ See [SL.str.5: Use `std::byte` to refer to byte values that do not necessarily r
 ### Non-member functions
 
 ```cpp
-template <class IntegerType, class = std::enable_if_t<std::is_integral<IntegerType>::value>>
+template <class IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value, bool> = true>
 constexpr byte& operator<<=(byte& b, IntegerType shift) noexcept;
 
-template <class IntegerType, class = std::enable_if_t<std::is_integral<IntegerType>::value>>
+template <class IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value, bool> = true>
 constexpr byte operator<<(byte b, IntegerType shift) noexcept;
 
-template <class IntegerType, class = std::enable_if_t<std::is_integral<IntegerType>::value>>
+template <class IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value, bool> = true>
 constexpr byte& operator>>=(byte& b, IntegerType shift) noexcept;
 
-template <class IntegerType, class = std::enable_if_t<std::is_integral<IntegerType>::value>>
+template <class IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value, bool> = true>
```

Comment:
I am hesitant to accept these operator changes because I don't believe they are "dangerous". From my experiments it is impossible to override the `std::enable_if_t` with an arbitrary type.

If I'm wrong, could you give me an example that compiles where the shift is a non-integer type?

Reply:
I started my changes with byte and added tests in `TEST(byte_tests, bitwise_operations )` like this:
```
operator<< <float,void>(a, float{1});
operator>> <float,void>(a, float{1});
operator<<= <float,void>(a, float{1});
operator>>= <float,void>(a, float{1});
```
These look totally strange, and they do trigger a compile error, but the error is inside the operator in byte. With the change the compile error is at the call site (no matching operator found).

Reply:
Ah! I didn't consider that. If these changes make compiler errors easier to understand, then I am all for it. I will re-evaluate with this point in mind.